packages/camera/camera_platform_interface/lib/src/events/camera_event.dart

```diff
@@ -235,3 +235,33 @@ class CameraErrorEvent extends CameraEvent {
   @override
   int get hashCode => super.hashCode ^ description.hashCode;
 }
+
+class CameraTimeLimitReachedEvent extends CameraEvent {
+  final XFile path;
```

Comment:
nit: Rename to `file` instead of `path`:
```suggestion
  final XFile file;
```